dist/comment_style.py

```diff
@@ -0,0 +1,143 @@
+                # So we can't just count the number of lines we need to check that there is no text
+                # preceding the comment in subsequent lines.
+                if length != 0 and not text_check.match(line.strip()):
+                    # If the comment is length 2 and we found another matching line then we have
+                    # found an illegal comment.
+                    if length == 2:
+                        print_msg(file_name, line_num, line, "multiline ")
+                        count += 1
+                    length += 1
+                    # Try and print only one error per comment, just keep incrementing the count and
+                    # the above if will only run once.
+                else:
+                    length += 1
+            else:
+                length = 0
+            line_num += 1
+        return count
+
+def file_is_cpp(name):
+    if re.search('(.cpp|.cxx|.hpp)$', name) is not None:
+        return True
+    if re.search('(.c|.i|.in)$', name) is not None:
+        return False
+
+    # House style is that C++ header files use ".h", which unfortunately makes
+    # this sort of code rather difficult. Luckily, libmagic can identify C/C++
+    # based on content. Don't import it because Python packaging is a disaster
+    # and this script needs to run reliably.
+    result = subprocess.run("file {}".format(name),
+                            shell=True,
+                            capture_output=True,
+                            text=True).stdout.strip('\n')
+    return "C++" in result
+
+# A collection of cases we handle.
+tests = [
+    (r'// hello world', True),
+    (r'     // hello world', True),
+    (r'hello world', False),
+    (r'printf("Hello, World!\n"); // prints hello world', True),
+    (r'String url = "http://www.example.com"', False),
+    (r'// hello world', True),
+    (r'//\\', True),
+    (r'// "some comment"', True),
+    (r'new URI("http://www.google.com")', False),
+    (r'printf("Escaped quote\""); // Comment', True),
+    (r' * http://www.google.com', False)
+]
+
+def validate(line, expect_match):
+    if (expect_match and not match(line.strip())) or (not expect_match and match(line.strip())):
+        print("Test failed:" + line)
+    else:
+        print("Test success")
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-t', '--test', default=False,
+                        help='Run self tests', action='store_true')
+    parser.add_argument('-f', '--fast', default=False,
+                        help='Use fast mode by only checking files changed in git',
+                        action='store_true')
+    args = parser.parse_args()
+
+    if args.test:
+        for test in tests:
+            validate(test[0], test[1])
+            sys.exit(0)
+
+    # Move up to root dir.
+    os.chdir("..")
+
+    # Some files aren't expected to comply with WiredTiger style. Ignore them.
+    ignore_files = [
```

Comment:
It looks like the `ignored_folders` list is now gone. Do we no longer need it?

Reply:
We found that we could fix most occurrences in the {{ignored_folders}} list. Its also a bit misleading as technically we only search a subset of the folders to begin with. i.e. `bench examples ext src test`

Reply:
Nope - this came out of Luke's comment here: https://github.com/wiredtiger/wiredtiger/pull/8629#discussion_r1061045551